certora/specs/ValidatorsExitBusOracle.spec

```diff
@@ -0,0 +1,355 @@
+
+    submitReportDataHelper@withrevert(e, consensusVersion, refSlot, requestsCount, dataFormat, dataInput, contractVersion);
+    bool isReverted = lastReverted;
+
+    uint256 totalRequestsProcessedAfter = getTotalRequestsProcessed();
+
+    assert requestsCount > max => isReverted;
+    assert requestsCount == 0 => totalRequestsProcessedBefore == totalRequestsProcessedAfter;
+    assert !isAllowed => isReverted;    // The caller is not a member of the oracle committee and doesn't possess the SUBMIT_DATA_ROLE.
+    assert contracrVersionReverted => isReverted;   // The provided contract version is different from the current one.
+    assert consensusVersionBefore != consensusVersion => isReverted;   // The provided consensus version is different from the expected one.
+    assert refSlotBefore != refSlot => isReverted;   // The provided reference slot is different from the expected one.
+    assert e.block.timestamp > deadline => isReverted; // The processing deadline for the current consensus frame is missed.
+}
+
+
+// STATUS - verified
+// no function, except submitReportData(), can change the value in LAST_PROCESSING_REF_SLOT_POSITION
+rule whoCanChangeLastProcessingEtc(env e, method f) filtered { f -> excludeInitialize(f) && excludeHelper(f) } {
+
+    uint256 getLastProcessingRefSlotBefore = getLastProcessingRefSlot();
+
+    calldataarg args;
+    f(e, args);
+
+    uint256 getLastProcessingRefSlotAfter = getLastProcessingRefSlot();
+
+    assert getLastProcessingRefSlotBefore != getLastProcessingRefSlotAfter
+            => f.selector == submitReportData((uint256,uint256,uint256,uint256,bytes),uint256).selector;
+}
+
+
+// STATUS - verified
+// After successfully processing a consensus report, the lastProcessingRefSlot must be updated correctly
+rule correctUpdateOfLastProcessingRefSlot() {
+    env e; env e2; env e3;
+
+    uint256 lastProcessingRefSlotBefore = getLastProcessingRefSlot();
+
+    uint256 contractVersion;
+    uint256 consensusVersion;   
+    uint256 refSlot;            
+    uint256 requestsCount;      
+    uint256 dataFormat;         
+    bytes   dataInput;          
+
+    require matchReportDatas(consensusVersion, refSlot, requestsCount, dataFormat, dataInput);
+    submitReportDataHelper(e, consensusVersion, refSlot, requestsCount, dataFormat, dataInput, contractVersion);
+
+    uint256 lastProcessingRefSlotAfter = getLastProcessingRefSlot();
+
+    assert lastProcessingRefSlotAfter == refSlot;
+}
+
+
+// STATUS - verified
+// getLastRequestedValidatorIndices() can't return the same index for the same nodeOpId for the same nodeOpIds
+rule noSameIndex(env e, method f) {
+    uint256 moduleId; 
+    uint256[] nodeOpIds;
+    
+    int256 indicesBefore1;
+    int256 indicesBefore2;
+    int256 indicesBefore3;
+
+    int256 indicesAfter1;
+    int256 indicesAfter2;
+    int256 indicesAfter3;
+
+    indicesBefore1, indicesBefore2, indicesBefore3 = callGetLastRequestedValidatorIndices(e, moduleId, nodeOpIds);
```

Comment:
Why don't you create a harness function that gives you any index?

Reply:
didn't think about it. will fix it